demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs

```diff
@@ -2,30 +2,43 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections;
+using System.Diagnostics;
 
 internal class Program
 {
     private static void Main(string[] args)
     {
-        if (args.Length == 1 && args[0] == "linux")
+        string osDetect = DetectOS();
+
+        if (args.Length >= 1)
+        {
+            osDetect = args[0];
+        }
+
+        if (osDetect == "linux")
```

Comment:
I'd suggest changing this to just be a switch statement now, especially with the duplicated DrawWindows at the end, e.g.
```C#
switch (osDetect)
{
    case "linux": 
        DrawLinux(); 
        break;
    case "freebsd": 
        DrawFreeBSD();
        break;
    case "mac":
        DrawMac();
        break;
    default:
        DrawWindows();
        break;
}
```